UglyTicTacToe/Default.aspx.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+using UglyTicTacToe.Models;
+using System.Windows;
+namespace UglyTicTacToe
+{
+    public partial class Default : System.Web.UI.Page
+    {
+        MyState lastState = Buttons.recentState;
+        MyState currentState;
```

Comment:
You can eliminate the `currentState` field and simply pass the "current state" into the `UpdateGameLogic` method via a parameter.